libs/ui/core/src/lib/brain/error-state-tracker.ts

```diff
@@ -0,0 +1,31 @@
+import { signal } from "@angular/core";
+import type { AbstractControl, FormGroupDirective, NgControl, NgForm } from "@angular/forms";
+import type { ErrorStateMatcher } from "./error-options";
+
+export class ErrorStateTracker {
+	/** Whether the tracker is currently in an error state. */
+	errorState = signal(false);
+
+	/** User-defined matcher for the error state. */
+	matcher: ErrorStateMatcher | null = null;
+
+	constructor(
```

Comment:
Let's use the inject function here

Reply:
I get your point, but this is not a service is just a class.